Less free IRAM after commit a1ac636

I can't compile after commit a1ac636:

`
/opt/esp-open-sdk/xtensa-lx106-elf/lib/gcc/xtensa-lx106-elf/4.8.5/../../../../xtensa-lx106-elf/bin/ld: out/build/app_0.out section .text' will not fit in regioniram1_0_seg'
/opt/esp-open-sdk/xtensa-lx106-elf/lib/gcc/xtensa-lx106-elf/4.8.5/../../../../xtensa-lx106-elf/bin/ld: region iram1_0_seg' overflowed by 24 bytes
collect2: error: ld returned 1 exit status
/opt/Sming/Sming/Makefile-rboot.mk:473: recipe for target 'out/build/app_0.out' failed
make: * [out/build/app_0.out] Error 1
`


As suggested by @slaff , I removed IRAM_ATTR from some methods in the HttpConnection and HttpServerConnection. After this, I've got "Free IRam : 3018" (with the latest version c224962)

--- a/Sming/SmingCore/Network/Http/HttpConnection.h
+++ b/Sming/SmingCore/Network/Http/HttpConnection.h
@@ -85,19 +85,19 @@ protected:
 	void cleanup();
 
 private:
-	static int IRAM_ATTR staticOnMessageBegin(http_parser* parser);
+	static int staticOnMessageBegin(http_parser* parser);
 #ifndef COMPACT_MODE
-	static int IRAM_ATTR staticOnStatus(http_parser *parser, const char *at, size_t length);
+	static int staticOnStatus(http_parser *parser, const char *at, size_t length);
 #endif
-	static int IRAM_ATTR staticOnHeadersComplete(http_parser* parser);
-	static int IRAM_ATTR staticOnHeaderField(http_parser *parser, const char *at, size_t length);
-	static int IRAM_ATTR staticOnHeaderValue(http_parser *parser, const char *at, size_t length);
-	static int IRAM_ATTR staticOnBody(http_parser *parser, const char *at, size_t length);
+	static int staticOnHeadersComplete(http_parser* parser);
+	static int staticOnHeaderField(http_parser *parser, const char *at, size_t length);
+	static int staticOnHeaderValue(http_parser *parser, const char *at, size_t length);
+	static int staticOnBody(http_parser *parser, const char *at, size_t length);
 #ifndef COMPACT_MODE
-	static int IRAM_ATTR staticOnChunkHeader(http_parser* parser);
-	static int IRAM_ATTR staticOnChunkComplete(http_parser* parser);
+	static int staticOnChunkHeader(http_parser* parser);
+	static int staticOnChunkComplete(http_parser* parser);
 #endif
-	static int IRAM_ATTR staticOnMessageComplete(http_parser* parser);
+	static int staticOnMessageComplete(http_parser* parser);
 
 	void sendRequestHeaders(HttpRequest* request);
 	bool sendRequestBody(HttpRequest* request);

--- a/Sming/SmingCore/Network/Http/HttpServerConnection.h
+++ b/Sming/SmingCore/Network/Http/HttpServerConnection.h
@@ -56,13 +56,13 @@ protected:
 	const char * getStatus(enum http_status s);
 
 private:
-	static int IRAM_ATTR staticOnMessageBegin(http_parser* parser);
-	static int IRAM_ATTR staticOnPath(http_parser *parser, const char *at, size_t length);
-	static int IRAM_ATTR staticOnHeadersComplete(http_parser* parser);
-	static int IRAM_ATTR staticOnHeaderField(http_parser *parser, const char *at, size_t length);
-	static int IRAM_ATTR staticOnHeaderValue(http_parser *parser, const char *at, size_t length);
-	static int IRAM_ATTR staticOnBody(http_parser *parser, const char *at, size_t length);
-	static int IRAM_ATTR staticOnMessageComplete(http_parser* parser);
+	static int staticOnMessageBegin(http_parser* parser);
+	static int staticOnPath(http_parser *parser, const char *at, size_t length);
+	static int staticOnHeadersComplete(http_parser* parser);
+	static int staticOnHeaderField(http_parser *parser, const char *at, size_t length);
+	static int staticOnHeaderValue(http_parser *parser, const char *at, size_t length);
+	static int staticOnBody(http_parser *parser, const char *at, size_t length);
+	static int staticOnMessageComplete(http_parser* parser);
 
 	void sendResponseHeaders(HttpResponse* response);
 	bool sendResponseBody(HttpResponse* response);

